docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -0,0 +1,1382 @@
+2021-06-24 18:14:22,102 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy]
+
+[ [ SubstrateSegfaultHandler caught a segfault. ] ]
+
+...
+----
+
+Now let’s try to debug the segmentation fault using `gdb`.
+We will start our application in `gdb` and execute `run`,
+then we will try to access http://localhost:8080/hello.
+
+[source,bash]
+----
+gdb ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner
+...
+Reading symbols from ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner...
+(No debugging symbols found in ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner)
+(gdb) run
+Starting program: /home/zakkak/tmp/code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner
+...
+curl http://localhost:8080/hello
+----
+
+This will result in the following message in `gdb`:
+
+[source,bash]
+----
+Thread 4 "ecutor-thread-1" received signal SIGSEGV, Segmentation fault.
+[Switching to LWP 693675]
+0x0000000000407380 in ?? ()
+Missing separate debuginfos, use: dnf debuginfo-install glibc-2.33-15.fc34.x86_64 sssd-client-2.5.0-2.fc34.x86_64 zlib-1.2.11-26.fc34.x86_64
+----
+
+If we try to get more info about the backtrace that led to this crash we will see that there is not enough information available.
+
+[source,bash]
+----
+(gdb) bt
+#0  0x0000000000418b5e in ?? ()
+#1  0x00007ffff6f2d328 in ?? ()
+#2  0x0000000000418a04 in ?? ()
+#3  0x00007ffff44062a0 in ?? ()
+#4  0x00000000010c3dd3 in ?? ()
+#5  0x0000000000000100 in ?? ()
+#6  0x0000000000000000 in ?? ()
+----
+
+This is because we didn’t compile the Quarkus application with `-Dquarkus.native.debug.enabled`,
+so `gdb` cannot find debugging symbols for our native executable,
+as indicated by the "_No debugging symbols found in ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner_" message in the beginning of `gdb`.
+
+Recompiling the Quarkus application with `-Dquarkus.native.debug.enabled` and rerunning it through `gdb` we are now able to get a backtrace making clear what caused the crash.
+On top of that, add `-H:-OmitInlinedMethodDebugLineInfo` option to avoid inlined methods being omitted from the backtrace:
+
+[source,bash,subs=attributes+]
+----
+./mvnw package -DskipTests -Pnative -Dquarkus.native.container-build=true \
+    -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel:{mandrel-flavor} \
+    -Dquarkus.native.debug.enabled \
+    -Dquarkus.native.additional-build-args=-H:-OmitInlinedMethodDebugLineInfo
+...
+$ gdb ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner
+Reading symbols from ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner...
+Reading symbols from /home/zakkak/tmp/code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner.debug...
+(gdb) run
+Starting program: /home/zakkak/tmp/code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner
+...
+$ curl http://localhost:8080/hello
+----
+
+This will result in the following message in `gdb`:
+
+[source,bash]
+----
+Thread 4 "ecutor-thread-0" received signal SIGSEGV, Segmentation fault.
```

Comment:
```suggestion
Thread 4 "executor-thread-0" received signal SIGSEGV, Segmentation fault.
```

Reply:
That's not a typo. Kernel thread names are of limited length and the start of Java thread names gets trimmed to fit.

Reply:
Interesting, thanks